### PR TITLE
Milestone 4 Calculator project

### PR DESCRIPTION
**Pull request by MFahadUmer**
- Added Calculate.js file
-- Calculate Method Accepts DataObject and Button name
-- It checks the Button name and then operates according to the operation.
- Added Operate.js fIle
-- Operate method accepts two numbers with the operation and returns the result

### CODE REVIEW

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import operate from './operate';
+
+const calculate = (dataObject, buttonName) => {
+  let { total, next, operation } = dataObject;
+  const buttonOperations = ['+', '-', 'X', '/', '%'];
```
Comment (whiz25) [inline src/logic/calculate.js]: - `+/-` is also one of the operations. add it to `buttonOperations`

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      } else {
+        operation = buttonName;
+      }
+    } else if (buttonName === '=/*') {
+      if (next) {
+        next = operate(next, '-1', 'X');
+      } else if (total) {
+        total = operate(total, '-1', 'X');
+      }
+    } else if (buttonName === '.') {
```
Comment (whiz25) [inline src/logic/calculate.js:37]: - check for `buttonName === '+/-` and if that is the case, multiply `total` and `next` by `-1` to make it negative or positive.

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  if (operation === '-') {
+    result = numberOneBig - numberTwoBig;
+  } else if (operation === '+') {
+    result = numberOneBig + numberTwoBig;
+  } else if (operation === 'X') {
+    result = numberOneBig * numberTwoBig;
+  } else if (operation === '/') {
+    result = numberOneBig / numberTwoBig;
+  } else if (operation === '%') {
+    result = (numberOneBig / numberTwoBig) * 100;
```
Comment (whiz25) [inline src/logic/operate.js]: - Use `numberOneBig.mod(numberTwoBig) * 100`. The `mod` function will handle for you the rounding of the result and it comes with the `Big.js` package.

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 
Awesome job in this PR. You managed to cover most of the requirements for this project. For example, your linter cases all pass.
But we need to fix some requirements to make sure we hit all the milestones.
- [ ] firstly, add a title to this PR. Learn how to do that [here](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request)
- [ ] read my comments for suggestions


### Martin Afani

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 
Nice job for making the requested changes. `+/-` has been added to the list of operations as requested. That's good.
You however, forgot to add a title to this PR. The title is a requirement for approval of this PR. Please add a title and resend for another review.


### Martin Afani

**Comment by MFahadUmer** [comment]
Thanks for the code review. Can you please explain how to add a title? Because already have a PR title (Milestone 4:Calculator Project). I will be very thankful if you can let me know the procedure.

**Review by rOluochKe — APPROVED** [review]
Hello @MFahadUmer 
### Well done, you got it right. The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one. 
- Happy coding!

**Comment by whiz25** [comment]
> Thanks for the code review. Can you please explain how to add a title? Because already have a PR title (Milestone 4:Calculator Project). I will be very thankful if you can let me know the procedure.

Learn how here:  https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request

**Pull request merged**